packages/mui-system/src/styleFunctionSx/defaultSxConfig.js

```diff
@@ -1,9 +1,42 @@
+import { unstable_capitalize as capitalize } from '@mui/utils';
 import { padding, margin } from '../spacing';
+import { handleBreakpoints } from '../breakpoints';
 import { borderRadius, borderTransform } from '../borders';
 import { gap, rowGap, columnGap } from '../cssGrid';
 import { paletteTransform } from '../palette';
 import { maxWidth, sizingTransform } from '../sizing';
 
+const createFontStyleFunction = (prop) => {
+  return (props) => {
+    if (props[prop] !== undefined && props[prop] !== null) {
+      const styleFromPropValue = (propValue) => {
+        let value =
+          props.theme.typography?.[
+            `${prop}${
+              props[prop] === 'default' || props[prop] === prop
```

Comment:
Just curious, what is this `props[prop] === prop` for? CSS property's value being equal to it's own property?